src/key.rs

```diff
@@ -449,6 +454,7 @@ impl PublicKey {
     }
 
     /// Creates a public key directly from a slice.
+    #[deprecated(since = "TBD", note = "Use `from_byte_array` instead.")]
```

Comment:
We must not deprecate this method because wanting to deserialize from a slice of either length is still a reasonable, valid operation that isn't solved by Rust's conversion traits and likely also not by slice patterns (yet) and even if it was, the error type is quite handy.

Reply:
Removed deprecation